paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -5,284 +5,267 @@ import android.view.LayoutInflater
 import com.stripe.android.model.PaymentMethod
-import com.stripe.android.model.StripeIntent
-import com.stripe.android.paymentsheet.databinding.FragmentPaymentsheetAddPaymentMethodBinding
+import com.stripe.android.paymentsheet.forms.FormFieldValues
+import com.stripe.android.paymentsheet.forms.FormViewModel.Companion.getElements
 import com.stripe.android.paymentsheet.model.PaymentSelection
 import com.stripe.android.paymentsheet.model.getPMAddForm
-import com.stripe.android.paymentsheet.paymentdatacollection.ComposeFormDataCollectionFragment
 import com.stripe.android.paymentsheet.paymentdatacollection.FormFragmentArguments
-import com.stripe.android.paymentsheet.paymentdatacollection.ach.USBankAccountFormFragment
+import com.stripe.android.paymentsheet.ui.Loading
+import com.stripe.android.paymentsheet.ui.PaymentMethodForm
 import com.stripe.android.paymentsheet.ui.PrimaryButton
 import com.stripe.android.paymentsheet.viewmodels.BaseSheetViewModel
-import com.stripe.android.ui.core.Amount
+import com.stripe.android.ui.core.FieldValuesToParamsMapConverter
 import com.stripe.android.ui.core.PaymentsTheme
-import com.stripe.android.ui.core.forms.resources.LpmRepository.SupportedPaymentMethod
-import com.stripe.android.utils.AnimationConstants
+import com.stripe.android.ui.core.elements.FormElement
+import com.stripe.android.ui.core.elements.IdentifierSpec
+import com.stripe.android.ui.core.forms.resources.LpmRepository
 import kotlinx.coroutines.flow.MutableStateFlow
-import kotlinx.coroutines.flow.first
-import kotlinx.coroutines.launch
 
 internal abstract class BaseAddPaymentMethodFragment : Fragment() {
     abstract val viewModelFactory: ViewModelProvider.Factory
     abstract val sheetViewModel: BaseSheetViewModel<*>
 
-    private lateinit var viewBinding: FragmentPaymentsheetAddPaymentMethodBinding
-    private var showLinkInlineSignup = MutableStateFlow(false)
-
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View? {
-        val themedInflater = inflater.cloneInContext(
-            ContextThemeWrapper(requireActivity(), R.style.StripePaymentSheetAddPaymentMethodTheme)
-        )
-        return themedInflater.inflate(
-            R.layout.fragment_paymentsheet_add_payment_method,
-            container,
-            false
-        )
-    }
+    ) = ComposeView(requireContext()).apply {
+        val elementsFlow = MutableStateFlow<List<FormElement>?>(null)
+        val showCheckboxFlow = MutableStateFlow(false)
 
-    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        viewBinding = FragmentPaymentsheetAddPaymentMethodBinding.bind(view)
+        setContent {
+            val isRepositoryReady by sheetViewModel.isResourceRepositoryReady.observeAsState(false)
+            val processing by sheetViewModel.processing.observeAsState(false)
```

Comment:
Can we pull this out into a Composable? Would make it easier in the future to get rid of `BaseAddPaymentMethodFragment` entirely.